graylog2-server/src/main/java/org/graylog2/configuration/ExposedConfiguration.java

```diff
@@ -0,0 +1,117 @@
+package org.graylog2.configuration;
+
+import com.google.common.collect.ImmutableList;
+import org.graylog2.Configuration;
+import org.graylog2.rest.models.system.configuration.ConfigurationVariable;
+
+/**
+ * List of configuration values that are safe to return, i.e. do not include any sensitive
+ * information. Building a list manually because we need to guarantee never to return any
+ * sensitive variables like passwords etc. - See this as a whitelist approach.
+ */
+public class ExposedConfiguration {
+
+    private final int inputBufferProcessors;
+    private final int processBufferProcessors;
+    private final int outputBufferProcessors;
+
+    private final String processorWaitStrategy;
+    private final String inputBufferWaitStrategy;
+    private final int inputBufferRingSize;
+    private final int ringSize;
+    private final String pluginDir;
+    private final String nodeIdFile;
+
+    private final boolean allowHighlighting;
+    private final boolean allowLeadingWildcardSearches;
+
+    private final String rotationStrategy;
+    private final String retentionStrategy;
+
+    private final int maxDocsPerIndex;
+    private final long maxSizePerIndex;
+    private final String maxTimePerIndex;
+    private final int maxNumberOfIndices;
+    private final int shards;
+    private final int replicas;
+
+    private final long streamProcessingTimeout;
+    private final int streamProcessingMaxFaults;
+    private final long outputModuleTimeout;
+    private final int staleMasterTimeout;
+
+    private final boolean disableIndexOptimization;
+    private final int indexOptimizationMaxSegments;
+
+    private final String gcWarningThreshold;
+
+    public ExposedConfiguration(Configuration configuration, ElasticsearchConfiguration esConfiguration) {
```

Comment:
What do we need `ConfigurationVariable` and `ConfigurationList` for if we could simply expose this class as JSON?